tools/platyPS/GenerateHelp.ps1

```diff
@@ -0,0 +1,133 @@
+
+    foreach ($file in $files)
+    {
+        # Ignore the module page
+        if ($file.Name -eq "AzureRM.$Service.md")
+        {
+            continue
+        }
+
+        $content = Get-Content $HelpPath\$file
+        $flag = $False
+
+        $newContent = New-Object string[] $content.Length
+
+        for ($idx = 0; $idx -lt $content.Length; $idx++)
+        {
+            if ($content[$idx] -like "*### -Confirm*" -and $content[$idx + 1] -like "*@{Text=}")
+            {
+                $flag = $True
+                $newContent[$idx] = $content[$idx]
+                $newContent[++$idx] = "Prompts you for confirmation before running the cmdlet."
+                    
+            }
+            elseif ($content[$idx] -like "*### -WhatIf*" -and $content[$idx + 1] -like "*@{Text=}")
+            {
+                $flag = $True
+                $newContent[$idx] = $content[$idx]
+                $newContent[++$idx] = "Shows what would happen if the cmdlet runs. The cmdlet is not run."
+            }
+            else
+            {
+                $newContent[$idx] = $content[$idx]
+            }
+        }
+
+        # Replace the contents of the markdown file if we found an error with the descriptions
+        if ($flag)
+        {
+            $result = $newContent -join "`r`n"
+            $tempFile = Get-Item $HelpPath\$file
+
+            [System.IO.File]::WriteAllText($tempFile.FullName, $result)
+        }
+    }
+}
+
+# Generate the MAML help file from the markdown files
+New-ExternalHelp -Path $HelpPath -OutputPath $PathToRepo\src\ResourceManager\$Service\Commands.$Service -Force
+
+# Get the content for the MAML file
+$content = Get-Content $PathToRepo\src\ResourceManager\$Service\Commands.$Service\Microsoft.Azure.Commands.$Service.dll-Help.xml
+
+# Keep track of the number of examples we find so we can add enough space in the new array
+$exampleCount = 0
+
+for ($idx = 0; $idx -lt $content.Length; $idx++)
+{
+    if ($content[$idx] -like "*<command:example>*")
+    {
+        $exampleCount++
+    }
+}
+
+# Since we will be adding two <maml:para></maml:para> tags to the MAML, we need to adjust the size of the array
+$newContentLength = $content.Length + (2 * $exampleCount)
+$newContent = New-Object string[] $newContentLength
+
+$buffer = 0
+
+for ($idx = 0; $idx -lt $content.Length; $idx++)
+{
+    $newContent[$idx + $buffer] = $content[$idx]
+
+    # If the next line is going to be the end of a remark in an example, add the two new lines for spacing
+    if ($content[$idx+1] -like "*</dev:remarks>*")
+    {
+        $newContent[$idx + $buffer + 1] = "<maml:para></maml:para>"
+        $newContent[$idx + $buffer + 2] = "<maml:para></maml:para>"
+
+        $buffer += 2
+    }
+}
+
+# Replace the contents of the current MAML with the new contents
+$result = $newContent -join "`r`n"
+$tempFile = Get-Item $PathToRepo\src\ResourceManager\$Service\Commands.$Service\Microsoft.Azure.Commands.$Service.dll-Help.xml
```

Comment:
It would be nice to have a chack here to ensure that the maml file was still valid xml.  You cna do this by using .Net classes, like XDocument.Load (which will throw if the document does not parse as correct xml).

The alternative would be to make these transformations using XPath, which would guarantee valid xml before and after.  I think havign a check at the end is fine for now.